src/SFA.DAS.Assessor.Functions/Domain/EpaoDataSync/EpaoDataSyncLearnerService.cs

```diff
@@ -0,0 +1,130 @@
+        {
+            try
+            {
+                _logger.LogDebug($"Using data collection api base address: {_dataCollectionServiceApiClient.BaseAddress()}");
+                _logger.LogDebug($"Using assessor api base address: {_assessorServiceApiClient.BaseAddress()}");
+
+                await ExportLearnerDetails(providerMessage);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, $"Epao data sync of Ukprn {providerMessage.Ukprn}, process learners failed for '{providerMessage.Source}' on page {providerMessage.LearnerPageNumber} ");
+                throw;
+            }
+        }
+
+        private async Task ExportLearnerDetails(EpaoDataSyncProviderMessage providerMessage)
+        {
+            var aimType = 1;
+            var allStandards = -1;
+            var fundModels = ConfigHelper.ConvertCsvValueToList<int>(_options.Value.LearnerFundModels);
+            var pageSize = _options.Value.LearnerPageSize;
+
+            var learnersPage = await _dataCollectionServiceApiClient.GetLearners(providerMessage.Source, providerMessage.Ukprn, aimType, allStandards, fundModels, pageSize, providerMessage.LearnerPageNumber);
+            if (learnersPage != null)
+            {
+                if (learnersPage.PagingInfo.TotalPages > learnersPage.PagingInfo.PageNumber)
+                {
+                    // queue a new message to process the next page
+                    await _epaoServiceBusQueueService.SerializeAndQueueMessage(
+                        new EpaoDataSyncProviderMessage 
+                        { 
+                            Ukprn = providerMessage.Ukprn, 
+                            Source = providerMessage.Source, 
+                            LearnerPageNumber = providerMessage.LearnerPageNumber + 1 
+                        });
+                }
+
+                // the learners must be filtered to remove learning deliveries which do not match the filters as the 
+                // data collection API returns learners which have at least one learning delivery matching a filter, 
+                // but it then returns ALL learning deliveries for each matching learner
+                var filteredLearners = FilterLearners(learnersPage.Learners, providerMessage.Source, aimType, fundModels);
+                if (filteredLearners.Count > 0)
+                {
+                    ImportLearnerDetailRequest request = new ImportLearnerDetailRequest
+                    {
+                        ImportLearnerDetails = filteredLearners
+                    };
+
+                    var response = await _assessorServiceApiClient.ImportLearnerDetails(request);
+                    response?.LearnerDetailResults?.ForEach(ld =>
+                    {
+                        ld?.Errors?.ForEach(e =>
+                        {
+                            _logger.LogDebug($"Unable to export Learner Details due to '{ld.Outcome}' '{e}'");
```

Comment:
we call **Import**LearnerDetails however this log line says unable to **Export**. Should these be consistent?

Reply:
I wondered about that when I wrote it I suppose 'request to import failed' would better...